tests/test_api_build.py

```diff
@@ -651,6 +651,26 @@ def test_noarch_python_1(testing_config):
     assert 'package_metadata_version' in extra
 
 
+def test_noarch_python2_only(testing_config):
+    output = api.build(os.path.join(metadata_dir, "_noarch_python2"), config=testing_config)[0]
+    assert package_has_file(output, 'info/files') is not ''
+    extra = json.loads(package_has_file(output, 'info/link.json').decode())
+    assert 'noarch' in extra
+    assert 'entry_points' in extra['noarch']
+    assert 'type' in extra['noarch']
+    assert 'package_metadata_version' in extra
```

Comment:
it would be good to add a step to these tests to try creating an env with both matching and mismatching python versions, to ensure that the desired constraint actually works.

Reply:
How would you propose doing that? FWICT `conda-build` just builds the recipes in this directory and tries to install them in their test environment. Seems like we have to change the testing infrastructure here without breaking the other recipes and how to do that is a bit unclear to me.

Reply:
use the environ.create_env function: https://github.com/conda/conda-build/blob/master/conda_build/environ.py#L737

Here's an example of it in use in a test: https://github.com/conda/conda-build/blob/e69e2b57cae76bfa4348565a18497c38a9bc0011/tests/test_environ.py#L16